src/main/scala/org/hammerlab/guacamole/commands/SomaticMutectLikeCaller.scala

```diff
@@ -0,0 +1,637 @@
+      }
+
+      if (args.cosmicVcf != "") {
+        val adamContext = new ADAMContext(sc)
+        val dbSnpVariants = adamContext.loadVariantAnnotations(args.cosmicVcf)
+        potentialGenotypes = potentialGenotypes
+          .keyBy(_.adamVariant)
+          .leftOuterJoin(dbSnpVariants.keyBy(_.getVariant))
+          .map(_._2).map({
+            case (calledAllele: CalledMutectSomaticAllele, cosmicVariant: Option[DatabaseVariantAnnotation]) =>
+              calledAllele.copy(cosOverlap = cosmicVariant.map(_ => true))
+          })
+      }
+
+      if (args.noiseVcf != "") {
+        val adamContext = new ADAMContext(sc)
+        val dbSnpVariants = adamContext.loadVariantAnnotations(args.cosmicVcf)
+        potentialGenotypes = potentialGenotypes
+          .keyBy(_.adamVariant)
+          .leftOuterJoin(dbSnpVariants.keyBy(_.getVariant))
+          .map(_._2).map({
+            case (calledAllele: CalledMutectSomaticAllele, noiseVariant: Option[DatabaseVariantAnnotation]) =>
+              calledAllele.copy(noiseOverlap = noiseVariant.map(_ => true))
+          })
+      }
+
+      val filteredGenotypes: RDD[CalledMutectSomaticAllele] = potentialGenotypes.filter(
+        finalMutectDbSnpCosmicNoisyFilter(_,
+          args.somDbSnpLODThreshold,
+          args.somNovelLODThreshold))
+
+      Common.progress("Computed %,d genotypes after basic filtering".format(filteredGenotypes.count))
+
+      Common.writeVariantsFromArguments(
+        args,
+        filteredGenotypes.flatMap(AlleleConversions.calledSomaticAlleleToADAMGenotype)
+      )
+
+      DelayedMessages.default.print()
+    }
+
+    def finalMutectDbSnpCosmicNoisyFilter(somAllele: CalledMutectSomaticAllele, somDbSnpThreshold: Double, somNovelThreshold: Double): Boolean = {
+      val recurrentMutSite = somAllele.cosOverlap.getOrElse(false)
+      val noisyMutSite = somAllele.noiseOverlap.getOrElse(false)
+      val dbSNPsite = somAllele.rsID.isDefined
+      val passSomatic: Boolean = ((dbSNPsite && somAllele.mutectEvidence.normalNotHet >= somDbSnpThreshold) ||
+        (!dbSNPsite && somAllele.mutectEvidence.normalNotHet >= somNovelThreshold))
+      val passNoiseFilter: Boolean = (recurrentMutSite || !noisyMutSite)
+
+      passSomatic && passNoiseFilter
+    }
+
+    def mapqBaseqFilteredPu(pu: Pileup, minBaseQuality: Int, minAlignmentQuality: Int): Pileup = {
+      Pileup(pu.referenceName,
+        pu.locus,
+        pu.referenceBase,
+        pu.elements.filter(pe =>
+          pe.qualityScore >= minBaseQuality && pe.read.alignmentQuality >= minAlignmentQuality))
+    }
+
+    def heavilyFilteredPu(pu: Pileup, maxFractionBasesSoftClippedTumor: Double, maxPhredSumMismatchingBases: Int): Pileup = {
+      Pileup(pu.referenceName,
+        pu.locus,
+        pu.referenceBase,
+        pu.elements.filterNot(pe => {
+          val clippedFilter = isReadHeavilyClipped(pe.read, maxFractionBasesSoftClippedTumor)
+          val noisyFilter = pe.read.mdTagOpt.map(m =>
```

Comment:
nit: intellij points out that `.map(…).getOrElse(false)` should be replaced with `.exists(…)`